pennylane/tape/tapes/jacobian_tape.py

```diff
@@ -239,22 +239,61 @@ def numeric_pd(self, idx, device, params=None, **options):
         shift[idx] = h
 
         if order == 1:
-            # Forward finite-difference.
-            # Check if the device has already be pre-computed with
-            # unshifted parameter values, to avoid redundant evaluations.
+            # forward finite-difference.
+
+            tapes = []
+
+            # get the stored result of the original circuit
             y0 = options.get("y0", None)
 
+            shifted = self.copy(deep=True, tape_cls=QuantumTape)
+            shifted.set_parameters(params + shift)
+
+            tapes.append(shifted)
+
             if y0 is None:
-                y0 = np.asarray(self.execute_device(params, device))
+                tapes.append(self)
```

Comment:
In this case, does `y0` ever stop being `None`?

Reply:
No, it doesn't, you are right. In most cases, `y0` should have been calculated before in the `jacobian` and added to the `options`. But sometimes `numeric_pd` gets called suddenly, for example as a fall-back method. This case is handled by adding the original tape to the `tapes` list, and changing the post-processing accordingly...

But since the `**options` is not a global data structure, adding `y0` to the options does